docs/DeveloperGuide.md

```diff
@@ -1,34 +1,845 @@
+
+    1.  To use **ATAS**, type a valid command into the terminal and press the enter key to run the command.  
+        e.g. Typing `help` and pressing the enter key will show the available commands
+
+    2.  Some example commands you can try to get familiar with **ATAS**:
+
+        -   `help`: Lists the commands that **ATAS** supports.
+
+        -   `assignment n/Assignment One m/CS2113T d/01/02/20 1600 c/Important Assignment`: Adds an assignment called **Assignment One** for the module **CS2113T**. This assignment is due on **1st February 2020 4pm** and the comments for this assignment is that it is an **Important Assignment**.
+
+        -   `exit`: Exits **ATAS**.
+
+## 2. Design
+This section will give a high-level overview of how various components in **ATAS** function and interact with each other.
+
+### 2.1. Architecture
+![overall architecture](images/overall architecture.PNG)
+
+The architecture diagram above illustrates the high-level design of the **ATAS** application.  
+
+The `Atas` component contains all the other components in the application.  
+
+-   `Ui`: Reads user input, and shows the results of commands to the user
+
+-   `Logic`: Handles execution of user input commands
+
+-   `Model`: Stores the runtime data of the application
+
+-   `Storage`: Reads and stores data from a file stored on the user’s computer
+
+The following sections will explain each component in greater detail.
+
+### 2.2. UI Component
+1.  The `Ui` component reads user input which represents a command.
+
+2.  The `Ui` component shows the result of the command to the user.
+
+### 2.3. Logic Component
+
+The Logic component comprises the `Parser`, `Command`, and `CommandResult` classes:  
+
+1.  `Parser` will interpret the user command and return the corresponding `Command` object.  
+
+2.  `Command#execute()` is called to run the command, returning a `CommandResult` object.  
+
+3.  The `CommandResult` object will contain the output to be shown to the user.
+
+### 2.4. Model Component
+The Model component contains the `Task` and `TaskList` classes, which store the user’s schedule.
+
+![TaskList and Tasks](images/TaskList Task class diagram.PNG)
+
+### 2.5. Storage Component
+![Storage Class Diagram](images/storage.PNG)
+
+1.  A `Storage` object is created by the `Atas` class to handle the loading and saving of `Task` data.
+
+2.  The `load()` method is used to read saved data from a local file into the current session of **ATAS**.
+
+3.  The `save()` method writes the current state of **ATAS** into the local save file using the `Task#encodeTask()` method.
+
+### 2.6. Atas Component
+The `Atas` component integrates all the aforementioned components to run the overall application logic.  
+The sequence diagram below shows how various components, broken down into the various classes, interact when the user enters a `help` command  
+
+![Component interactions for help command](images/atas help command sequence diagram v3.PNG)
```

Comment:
I think your UI activation bar should not have a break in between? Because it is waiting for the return result from "help".